File: app/src/main/java/maderski/chargingindicator/Services/CIService.java
package maderski.chargingindicator.Services;

import android.app.Service;
import android.content.ComponentName;
import android.content.Intent;
import android.content.IntentFilter;
import android.content.pm.PackageManager;
import android.os.IBinder;
import android.util.Log;
import android.widget.Toast;

import com.crashlytics.android.Crashlytics;

import io.fabric.sdk.android.Fabric;
import maderski.chargingindicator.Actions.AsyncConnectedActions;
import maderski.chargingindicator.Battery.BatteryManager;
import maderski.chargingindicator.BuildConfig;
import maderski.chargingindicator.Receivers.PowerConnectionReceiver;
import maderski.chargingindicator.Utils.ServiceUtils;

/**
 * Created by Jason on 12/6/15.
 */
public class CIService extends Service {

    private static final String TAG = CIService.class.getName();

    private final PowerConnectionReceiver mPowerConnectionReceiver = new PowerConnectionReceiver();

    //Instantiation of PowerConnectionReceiver and Registers receiver for ACTION_BATTERY_CHANGED
    @Override
    public int onStartCommand(Intent intent, int flags, int startId){

        Fabric.with(this, new Crashlytics());

        IntentFilter filter = new IntentFilter();
        filter.addAction("android.intent.action.ACTION_POWER_CONNECTED");
        filter.addAction("android.intent.action.ACTION_POWER_DISCONNECTED");

        registerReceiver(mPowerConnectionReceiver, filter);

        if(BuildConfig.DEBUG) {
            Log.i(TAG, "CIService Started");
            Toast.makeText(this, "CIService started", Toast.LENGTH_SHORT).show();
        }

        return Service.START_STICKY;
    }

    @Override
    public void onDestroy(){
        super.onDestroy();
        if(BuildConfig.DEBUG)
            Log.i(TAG, "Service stopped");

        unregisterReceiver(mPowerConnectionReceiver);
    }

    @Override
    public IBinder onBind(Intent intent){
        return null;
    }
}

File: app/src/main/java/maderski/chargingindicator/services/CIService.kt
package maderski.chargingindicator.services

import android.app.Service
import android.content.Intent
import android.content.IntentFilter
import android.os.Binder
import android.os.Build
import android.os.IBinder
import maderski.chargingindicator.R
import maderski.chargingindicator.receivers.PowerConnectionReceiver
import maderski.chargingindicator.utils.ServiceUtils

class CIService : Service() {
    private val cIBinder = CIBinder()
    private val powerConnectionReceiver = PowerConnectionReceiver()

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        val intentFilter = IntentFilter()
        intentFilter.addAction("android.intent.action.ACTION_POWER_CONNECTED")
        intentFilter.addAction("android.intent.action.ACTION_POWER_DISCONNECTED")

        registerReceiver(powerConnectionReceiver, intentFilter)

        return START_STICKY
    }

    override fun onCreate() {
        super.onCreate()
        val title = this.getString(R.string.ci_service_notification_title)
        val message = this.getString(R.string.ci_service_notification_messge)
        ServiceUtils.createServiceNotification(ServiceUtils.FOREGROUND_NOTIFICATION_ID,
                title,
                message,
                this,
                getString(R.string.ci_channel_id),
                getString(R.string.ci_channel_name),
                R.drawable.ic_action_battery,
                false)

    }

    override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(powerConnectionReceiver)
    }

    override fun onBind(intent: Intent?): IBinder? = cIBinder

    inner class CIBinder : Binder() {
        fun getService(): CIService = this@CIService
    }

    companion object {
        const val TAG = "CIService"
    }

}